systemtests/src/test/java/io/enmasse/systemtest/common/api/AddressControllerApiTest.java

```diff
@@ -0,0 +1,132 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@Category(IsolatedAddressSpace.class)
+public class AddressControllerApiTest extends TestBase {
+    private static Logger log = CustomLogger.getLogger();
+
+    @Override
+    protected String getDefaultPlan(AddressType addressType) {
+        return null;
+    }
+
+    @Before
+    public void setUp() {
+        plansProvider.setUp();
+    }
+
+    @After
+    public void tearDown() {
+        plansProvider.tearDown();
+    }
+
+    @Test
+    public void testRestApiGetSchema() throws Exception {
+        AddressPlan queuePlan = new AddressPlan("test-schema-rest-api-addr-plan", AddressType.QUEUE,
+                Collections.singletonList(new AddressResource("broker", 0.6)));
+        plansProvider.createAddressPlanConfig(queuePlan);
+
+        //define and create address space plan
+        List<AddressSpaceResource> resources = Arrays.asList(
+                new AddressSpaceResource("broker", 0.0, 2.0),
+                new AddressSpaceResource("router", 1.0, 1.0),
+                new AddressSpaceResource("aggregate", 0.0, 2.0));
+        List<AddressPlan> addressPlans = Collections.singletonList(queuePlan);
+        AddressSpacePlan addressSpacePlan = new AddressSpacePlan("schema-rest-api-plan", "schema-rest-api-plan",
+                "standard-space", AddressSpaceType.STANDARD, resources, addressPlans);
+        plansProvider.createAddressSpacePlanConfig(addressSpacePlan);
+
+        Future<SchemaData> data = getSchema();
+        SchemaData schemaData = data.get(20, TimeUnit.SECONDS);
+        log.info("Check if schema object is not null");
+        assertThat(schemaData.getAddressSpaceTypes().size(), not(0));
+
+        log.info("Check if schema object contains new address space plan");
+        assertTrue(schemaData.getAddressSpaceType("standard").getPlans()
+                .stream()
+                .map(PlanData::getName)
+                .collect(Collectors.toList()).contains("schema-rest-api-plan"));
+
+        log.info("Check if schema contains new address plans");
+        assertTrue(schemaData.getAddressSpaceType("standard").getAddressType("queue").getPlans().stream()
+                .filter(s -> s.getName().equals("test-schema-rest-api-addr-plan"))
+                .map(PlanData::getName)
+                .collect(Collectors.toList())
+                .contains("test-schema-rest-api-addr-plan"));
+    }
+
+    @Test
+    public void testRestApiAddressResourceOptionalParams() throws Exception {
+        AddressSpace addressSpace = new AddressSpace("test-rest-api-addr-space", AddressSpaceType.BROKERED);
+        createAddressSpace(addressSpace, AuthService.NONE.toString());
+
+        AddressSpace addressSpace2 = new AddressSpace("test-rest-api-addr-space2", AddressSpaceType.BROKERED);
+        createAddressSpace(addressSpace2, AuthService.NONE.toString());
+
+        log.info("--------------------------------------------------");
```

Comment:
could we create new method for this separator in TestBase? It could be useful in multiple tests, I would say